source/common/ratelimit/ratelimit_impl.cc

```diff
@@ -66,14 +67,18 @@ void GrpcClientImpl::onSuccess(
     span.setTag(Constants::get().TraceStatus, Constants::get().TraceOk);
   }
 
-  callbacks_->complete(status);
+  Http::HeaderMapPtr headers = std::make_unique<Http::HeaderMapImpl>();
```

Comment:
Can we optimize the case where there are no headers to add and pass nullptr here?